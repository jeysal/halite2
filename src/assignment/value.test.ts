import { Action } from 'halite/Game';
import GameMap from 'halite/GameMap';

import Goal from '../goal/goal';

import generateShipGoalValues from './value';

const goalStub: Goal = {
  value: () => 0.5,
  minShips: () => 0,
  maxShips: () => 16,
  suitability: () => 0.5,
  strive: () => '' as string & Action,
};

test('generates a value for each ship-goal combination', () => {
  const gm = new GameMap({ myPlayerId: 0, width: 100, height: 100 });
  gm.addPlayerShips(0, [{ id: 0 }, { id: 1 }, { id: 2 }]);

  const goal1: Goal = { ...goalStub };
  const goal2: Goal = { ...goalStub };

  expect(
    generateShipGoalValues([goal1, goal2], gm).map(goalValues =>
      goalValues.map(({ goal, ...value }) => value),
    ),
  ).toMatchSnapshot();
});

test('considers both value and suitability', () => {
  const gm = new GameMap({ myPlayerId: 0, width: 100, height: 100 });
  gm.addPlayerShips(0, [{ id: 0 }, { id: 1 }, { id: 2 }]);

  const goal1: Goal = {
    ...goalStub,
    suitability: ({ id }) => [0.1, 0.5, 0.9][id],
  };
  const goal2: Goal = {
    ...goalStub,
    suitability: ({ id }) => [0.3, 0.7, 0.5][id],
  };

  expect(
    generateShipGoalValues([goal1, goal2], gm).map(goalValues =>
      goalValues.map(({ goal, ...value }) => value),
    ),
  ).toMatchSnapshot();
});
