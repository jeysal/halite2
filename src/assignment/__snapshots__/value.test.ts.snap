// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`considers both value and suitability 1`] = `
Array [
  Array [
    Object {
      "shipId": 0,
      "val": 0.3,
    },
    Object {
      "shipId": 0,
      "val": 0.4,
    },
  ],
  Array [
    Object {
      "shipId": 1,
      "val": 0.5,
    },
    Object {
      "shipId": 1,
      "val": 0.6,
    },
  ],
  Array [
    Object {
      "shipId": 2,
      "val": 0.7,
    },
    Object {
      "shipId": 2,
      "val": 0.5,
    },
  ],
]
`;

exports[`generates a value for each ship-goal combination 1`] = `
Array [
  Array [
    Object {
      "shipId": 0,
      "val": 0.5,
    },
    Object {
      "shipId": 0,
      "val": 0.5,
    },
  ],
  Array [
    Object {
      "shipId": 1,
      "val": 0.5,
    },
    Object {
      "shipId": 1,
      "val": 0.5,
    },
  ],
  Array [
    Object {
      "shipId": 2,
      "val": 0.5,
    },
    Object {
      "shipId": 2,
      "val": 0.5,
    },
  ],
]
`;
